pkg/storaged/btrfs/subvolume.jsx

```diff
@@ -44,19 +47,137 @@ function subvolume_mount(volume, subvol, forced_options) {
     mounting_dialog(client, block, "mount", forced_options, subvol);
 }
 
+function get_mount_point_in_parent(volume, subvol) {
+    const block = client.blocks[volume.path];
+    const subvols = client.uuids_btrfs_subvols[volume.data.uuid];
+    if (!subvols)
+        return null;
+
+    for (const p of subvols) {
+        const has_parent_subvol = subvol.pathname.substring(0, p.pathname.length) == p.pathname &&
+                                 subvol.pathname[p.pathname.length] == "/";
+        if ((subvol.parent == p.id || has_parent_subvol) && is_mounted(client, block, p)) {
```

Comment:
The `subvol.parent == p.id` check only catches direct parents, and we also want indirect ones. I think we don't need this check since we always have the pathname.

Reply:
I added this for the `p.pathname == '/'` conditional which was here before which squints: might be fixed by:
```
(p.pathname == "/" && subvol.pathname  !== "/"
```

Reply:
So I found it interesting that the deletion code has:

```
 216   │     function get_direct_subvol_children(subvol) {
 217   │         function is_direct_parent(sv) {
 218   │             return (sv.pathname.length > subvol.pathname.length &&
 219   │                     sv.pathname.substring(0, subvol.pathname.length) == subvol.pathname &&
 220   │                     sv.pathname[subvol.pathname.length] == "/" &&
 221   │                     sv.pathname.substring(subvol.pathname.length + 1).indexOf("/") == -1);
 222   │         }
 223   │
 224   │         return subvols.filter(is_direct_parent);
```

So obviously that should be combined / re-used here. Maybe that way we can make it a pure function, write a good unit test. The only downside is that we iterate twice but that can't be too slow in JS.

Reply:
Nevermind that is something completely different

Reply:
Re-read it, applied your suggestion.